sdk/eventhub/Azure.Messaging.EventHubs/stress/src/Infrastructure/EventTracking.cs

```diff
@@ -0,0 +1,273 @@
+using System.Collections.Concurrent;
+using System.Collections.Generic;
+using System.Security.Cryptography;
+using Azure.Messaging.EventHubs.Consumer;
+
+namespace Azure.Messaging.EventHubs.Stress;
+
+/// <summary>
+///   A class that allows for keeping track of events as they are sent from one of the producer clients
+///   until they are received by the processor or consumer client. It is used by the producer client to
+///   add property values to the <see cref="EventData"/>. It is used by the processors and consumers
+///   to keep track of previously read events and determine if events being processed are out of order,
+///   duplicated, or have corrupted bodies.
+/// </summary>
+///
+public static class EventTracking
+{
+    /// <summary>
+    ///   The name of the <see cref="EventData"/> property that holds the producer-assigned index number.
+    /// </summary>
+    ///
+    public static readonly string IndexNumberPropertyName = "IndexNumber";
+
+    /// <summary>
+    ///   The name of the <see cref="EventData"/> property that holds the gtime the event was published.
+    /// </summary>
+    ///
+    public static readonly string PublishTimePropertyName = "PublishTime";
+
+    /// <summary>
+    ///   The name of the <see cref="EventData"/> property that holds the partition the producer
+    ///   was intending to send to.
+    /// </summary>
+    ///
+    public static readonly string PartitionPropertyName = "Partition";
+
+    /// <summary>
+    ///   The name of the <see cref="EventData"/> property that holds the ID assigned to this event by
+    ///   the producer.
+    /// </summary>
+    ///
+    public static readonly string IdPropertyName = "Identifier";
+
+    /// <summary>
+    ///   The name of the <see cref="EventData"/> property that holds the hash of the event body assigned by the producer.
+    /// </summary>
+    ///
+    public static readonly string EventBodyHashPropertyName = "EventDataHash";
+
+    /// <summary>
+    ///   Adds properties to the given <see cref="EventData"/> instance, allowing for the processor to determine that events
+    ///   were received in order, not duplicated, and have valid event bodies.
+    /// </summary>
+    ///
+    /// <param name="eventData">The <see cref="EventData"/> instance to augment.</param>
+    /// <param name="indexNumber">The producer assigned index number for this event.</param>
+    /// <param name="partition">The partition, if any, that this event was intended to be sent to.</param>
+    ///
+    public static void AugmentEvent(EventData eventData,
+                                    int indexNumber,
+                                    string partition=null)
+    {
+        eventData.Properties.Add(IndexNumberPropertyName, indexNumber);
+        eventData.Properties.Add(PublishTimePropertyName, DateTimeOffset.UtcNow);
+        eventData.Properties.Add(IdPropertyName, Guid.NewGuid().ToString());
+
+        using (SHA256 sha256Hash = SHA256.Create())
```

Comment:
Should this be using the `TestConfiguration.Sha256Hash` member?   Maybe by passing it in like `CheckEvent`?